browser/notifications/BUILD.gn

```diff
@@ -15,6 +15,7 @@ source_set("notifications") {
   deps = [
     "//base",
     "//brave/components/brave_adaptive_captcha/buildflags",
+    "//brave/vendor/bat-native-ads",
```

Comment:
It is used because it includes `ads_service_factory.h` and it includes `ads_service.h`. 
Seems like this could be changed to `"//brave/components/brave_ads/browser"` dep.